annotations/text_upload.py

```diff
@@ -0,0 +1,94 @@
+
+from annotations.models import Text, TextCollection, RelationSet
+from repository.models import Repository
+from repository.managers import RepositoryManager
+from django.shortcuts import get_object_or_404, render
+from django.http import HttpResponse
+
+
+
+
+
+def repository_text_content(user, repository_id, text_id, content_id, part_of_id, action, project_id):
+    repository = get_object_or_404(Repository, pk=repository_id)
+
+    manager = RepositoryManager(repository.configuration, user=user)
+    # content_resources = {o['id']: o for o in resource['content']}
+    # content = content_resources.get(int(content_id))    # Not a dict.
+    try:
+        content = manager.content(id=int(content_id))
+        resource = manager.resource(id=int(text_id))
+    except IOError:
+        return HttpResponse("IO Error", content_type="text/plain")
```

Comment:
why do you return an httpresponse object here?

Reply:
wait, where is this called? is this a view? i'm confused.

Reply:
Just like the above comment it started life as a view and has since moved onto greater things in the form of a helper function. I missed the httpResponses, I will fix them with more appropriate solutions.